src/meta/src/hummock/manager/mod.rs

```diff
@@ -997,6 +1011,18 @@ impl HummockManager {
             compact_task.current_epoch_time = Epoch::now().0;
             compact_task.compaction_filter_mask =
                 group_config.compaction_config.compaction_filter_mask;
+            table_to_vnode_partition
+                .retain(|table_id, _| compact_task.existing_table_ids.contains(table_id));
+
+            if group_config.compaction_config.split_weight_by_vnode > 0 {
+                // TODO: unify to table_to_vnode_partition directly
+                table_to_vnode_partition.insert(
+                    member_table_ids[0],
```

Comment:
Instead of updating table_to_vnode_partition ad-hoc in `get_compact_task`, I think it is more appropriate to update `self.group_to_table_vnode_partition` correspondingly in the following cases:
1. A new group is created.
2. `split_weight_by_vnode` is modified for an existing group.
3. New table ids are added to a group.

Maybe that is what you mean by "// TODO: unify to table_to_vnode_partition directly"?